platform/ui/src/components/userPreferencesForm/WindowLevelPreferences.js

```diff
@@ -1,94 +1,105 @@
-import React, { Component } from 'react';
+import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
-import './WindowLevelPreferences.styl';
 
-export class WindowLevelPreferences extends Component {
-  constructor(props) {
-    super(props);
+import './WindowLevelPreferences.styl';
 
-    this.state = {
-      data: this.props.windowLevelData,
-    };
-  }
+function WindowLevelPreferencesRow({
+  description,
+  window,
+  level,
+  rowName,
+  onSuccessChanged,
+  onFailureChanged,
```

Comment:
Same goes here for onFailureChanged not being used, we can keep it out for now to avoid warnings and also add jsDocs to this component like HotKeyPreferences.